packages/editor-ui/src/components/TextWithHighlights.vue

```diff
@@ -0,0 +1,39 @@
+<script lang="ts" setup>
+import type { PropType } from 'vue';
+import { GenericValue } from 'n8n-workflow';
+
+const props = defineProps({
+	content: {
+		type: [Object, String, Number] as PropType<GenericValue>,
+	},
+	search: {
+		type: String,
+	},
+});
+
+const splitTextBySearch = (text = '', search = ''): { tag: 'span' | 'mark'; content: string }[] => {
+	if (!search) {
+		return [
+			{
+				tag: 'span',
+				content: text,
+			},
+		];
+	}
+
+	const pattern = new RegExp(`(${search})`, 'g');
+	const splitText = text.split(new RegExp(pattern, 'gi'));
+
+	return splitText.map((t) => ({ tag: pattern.test(t) ? 'mark' : 'span', content: t }));
```

Comment:
I [previously](https://github.com/n8n-io/n8n/blob/master/packages/nodes-base/nodes/PostBin/GenericFunctions.ts#L22) had a problem with using regexes in a loop without resetting it's `lastIndex` to 0. Couldn't reproduce this here but just a heads up if you can think about cases where this could fail.

Reply:
thanks

Reply:
just removed the global flag... it's not needed with split